EDEngineer.Models/Operations/UpgradeOperation.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EDEngineer.Models.Operations
+{
+	public class UpgradeOperation : JournalOperation
+    {
+        private Dictionary<string, int> _changes = new Dictionary<string, int>();
+
+        public JournalEvent Event { get; set; }
+
+        public string Name { get; set; }
+
+		public int Class { get; set; }
+
+		public override Dictionary<string, int> Changes => throw new NotImplementedException();
```

Comment:
might want to return the `_changes` field here

Reply:
Done